src/arch/pmp.rs

```diff
@@ -219,8 +225,28 @@ impl PmpGroup {
         pmp
     }
 
+    pub fn set_napot(&mut self, idx: usize, from: usize, to: usize, permissions: u8) {
+        assert!(
+            permissions < 8,
+            "Permissions should not set NAPOT or TOP bits"
+        );
+        self.set(idx, build_napot(from, to).unwrap(), permissions | NAPOT);
+    }
+
+    pub fn set_tor(&mut self, idx: usize, until: usize, permissions: u8) {
+        assert!(
+            permissions < 8,
+            "Permissions should not set NAPOT or TOP bits"
+        );
+        self.set(idx, build_tor(until), permissions | TOR);
+    }
+
+    pub fn set_inactive(&mut self, idx: usize, addr: usize) {
+        self.set(idx, build_tor(addr), INACTIVE);
+    }
```

Comment:
Those functions are cool, but they lack some good documentation :)
They will be used in a bunch of places, so having a good doc comment would be great. Can you add an explanation for each of those?

Reply:
Done!